source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -486,26 +500,24 @@ absl::optional<QueuedChunkPtr> ChunkQueue::pop(bool undelivered_only) {
   }
   QueuedChunkPtr chunk = std::move(queue_.front());
   queue_.pop_front();
-  bytes_enqueued_ -= chunk->data.length();
+  bytes_enqueued_ -= chunk->length;
+
+  // Move the corresponding data out.
+  out_data.move(received_data_, chunk->length);
   return chunk;
 }
 
-const QueuedChunk& ChunkQueue::consolidate(bool delivered) {
-  if (queue_.size() == 1) {
-    queue_.front()->delivered = delivered;
-    return *(queue_.front());
+const QueuedChunk& ChunkQueue::consolidate() {
+  if (queue_.size() > 1) {
+    auto new_chunk = std::make_unique<QueuedChunk>();
```

Comment:
since we are refactoring code here, I am wondering if we can clean up code a bit
Iadd a constructor to `QueuedChunk` to init end_stream and length together.

```
queue_.emplace_front(make_unique<QueuedChunk>(queue_.back()->end_stream, bytes_enqueued_))
```

Basically we can avoid updating `end_stream` and `length` in separate lines in multiple places.

Reply:
Thanks for the comments!
Same as above. Let's fix the functionality issues first, and leaving the non-functional cleanup work later.

Reply:
SGTM, Thanks